docs/blog/2025-05-08-mopro-native-packages.md

```diff
@@ -0,0 +1,119 @@
+
+### Baseline Size
+
+The full build creates an archive around **≈ 153 MB** in size. When uploading files over 100 MB to GitHub, Git LFS takes over by replacing the file with a text pointer in the repository while storing the actual content on a remote server like GitHub.com. This setup can cause issues for package managers that try to fetch the package directly from a GitHub URL for a release publish.
+
+While uploading large files may be acceptable for some package management platforms or remote servers like Cloudflare R2, the large file size slows down:
+
+- CocoaPods or SwiftPM downloads
+- CI cache recovery
+- Cloning the repository, especially on slower connections
+
+### Our Solution: Zip & Unzip Strategy
+
+To keep development fast and responsive, we compress the entire `MoproBindings.xcframework` before uploading it to GitHub and publishing it to CocoaPods, reducing its size to about **≈ 41 MB**.
+
+We also found that by customizing `script_phase` in the `.podspec`, we can unzip the bindings during pod install. This gives us the best of both worlds: (1) smaller packages for distribution and (2) full compatibility with Xcode linking. The added CPU cost is minor compared to the time saved on downloads.
+
+### Comparison With Android
+
+On Android, dynamic `.so` libraries (around 22 MB in total) are used, with symbols loaded lazily at runtime to keep the package size small. In contrast, because iOS's constraint on third-party Rust dynamic libraries in App Store builds, static linking with compression is currently the most viable option, to the best of our knowledge.
+
+## The Road Ahead
+
+### a. Manual Tweaks for Cross-Platform Frameworks
+
+Cross-platform frameworks like React Native and Flutter require additional glue code to define modules, as they straddle multiple runtimes. Each layer needs its own integration. For example, in our zkEmail React Native package, we use three separate wrappers.
+
+- \[TypeScript\] [`MoproReactNativePackageModule.ts`](https://github.com/zkmopro/zkemail-react-native-package/blob/main/src/MoproReactNativePackageModule.ts): declares the public API and lets the React Native code-gen load the native module.
+- \[Swift\] [`MoproReactNativePackageModule.swift`](https://github.com/zkmopro/zkemail-react-native-package/blob/main/ios/MoproReactNativePackageModule.swift): loads bindings into Objective-C–discoverable classes.
+- \[Kotlin\] [`MoproReactNativePackageModule.kt`](https://github.com/zkmopro/zkemail-react-native-package/blob/main/android/src/main/java/expo/modules/moproreactnativepackage/MoproReactNativePackageModule.kt): loads bindings and bridges via JNI.
+
+### b. Support for Custom Package Names
+
+Initially, we encountered naming conflicts with the same XCFramework used in multiple Xcode projects. We addressed this with recent updates ([issue#387](https://github.com/zkmopro/mopro/issues/387), [PR#404](https://github.com/zkmopro/mopro/pull/404)), allowing customizable package names.
```

Comment:
we haven't fixed :(
#404 is just a part of it
now I created another issue: https://github.com/zkmopro/mopro/issues/413
once it is all fixed, then it should be done